hikari/impl/rest.py

```diff
@@ -1389,23 +1392,36 @@ def _build_message_payload(  # noqa: C901- Function too complex
         body.put("embeds", serialized_embeds)
         body.put("components", serialized_components)
 
-        if replace_attachments:
-            body.put("attachments", None)
-
         if not edit or not undefined.all_undefined(mentions_everyone, mentions_reply, user_mentions, role_mentions):
             body.put(
                 "allowed_mentions",
                 mentions.generate_allowed_mentions(mentions_everyone, mentions_reply, user_mentions, role_mentions),
             )
 
+        form_builder: typing.Optional[data_binding.URLEncodedFormBuilder] = None
         if final_attachments:
-            form_builder = data_binding.URLEncodedFormBuilder(executor=self._executor)
+            attachments_payload = []
+            attachment_id = 0
+
+            for f in final_attachments:
+                if isinstance(f, messages_.Attachment):
```

Comment:
You prob need to update the interaction server's logic as well

Reply:
Good spot! Forgot about that

Reply:
Implemented!